src/devices/Media/SoundDevice/Devices/UnixSoundDevice.cs

```diff
@@ -0,0 +1,553 @@
+        private IntPtr _mixer;
+        private IntPtr _elem;
+        private int _errorNum;
+
+        private static readonly object playbackInitializationLock = new object();
+        private static readonly object recordingInitializationLock = new object();
+        private static readonly object mixerInitializationLock = new object();
+
+        /// <summary>
+        /// The connection settings of the sound device.
+        /// </summary>
+        public override SoundConnectionSettings Settings { get; }
+
+        /// <summary>
+        /// The playback volume of the sound device.
+        /// </summary>
+        public override long PlaybackVolume
+        {
+            get => GetPlaybackVolume();
+            set
+            {
+                SetPlaybackVolume(value);
+            }
+        }
+
+        // The lib do not have a method of get all channels mute state.
+        private bool _playbackMute;
+        /// <summary>
+        /// The playback mute of the sound device.
+        /// </summary>
+        public override bool PlaybackMute
+        {
+            get => _playbackMute;
+            set
+            {
+                SetPlaybackMute(value);
+                _playbackMute = value;
+            }
+        }
+
+        /// <summary>
+        /// The recording volume of the sound device.
+        /// </summary>
+        public override long RecordingVolume { get => GetRecordingVolume(); set => SetRecordingVolume(value); }
+
+        private bool _recordingMute;
+        /// <summary>
+        /// The recording mute of the sound device.
+        /// </summary>
+        public override bool RecordingMute
+        {
+            get => _recordingMute;
+            set
+            {
+                SetRecordingMute(value);
+                _recordingMute = value;
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UnixSoundDevice"/> class that will use the specified settings to communicate with the sound device.
+        /// </summary>
+        /// <param name="settings">The connection settings of a sound device.</param>
+        public UnixSoundDevice(SoundConnectionSettings settings)
+        {
+            Settings = settings;
+
+            PlaybackMute = false;
+            RecordingMute = false;
+        }
+
+        /// <summary>
+        /// Play WAV file.
+        /// </summary>
+        /// <param name="wavPath">WAV file path.</param>
+        public override void Play(string wavPath)
+        {
+            using FileStream fs = File.Open(wavPath, FileMode.Open);
+
+            Play(fs);
+        }
+
+        /// <summary>
+        /// Play WAV file.
+        /// </summary>
+        /// <param name="wavStream">WAV stream.</param>
+        public override void Play(Stream wavStream)
+        {
+            IntPtr @params = new IntPtr();
+            int dir = 0;
+            WavHeader header = ReadWavHeader(wavStream);
+
+            OpenPlaybackPcm();
```

Comment:
please put these in `try`..`finally` so that you ensure closing the device in case something goes wrong

Reply:
The `ThrowIfError` method will call `Dispose` when an exception occurs internally. I think this step is not necessary. But I will add `try...finally`.